File: example2.cs
using System;
using System.Collections.Generic;

class Program
{
    public static int length(string s)
    {
        int maxLength = 0;
        int left = 0;
        Dictionary<char, int> lastIndex = new Dictionary<char, int>();

        for (int right = 0; right < s.Length; right++)
        {
            if (lastIndex.ContainsKey(s[right]) && lastIndex[s[right]] >= left)
            {
                left = lastIndex[s[right]] + 1;
            }

            lastIndex[s[right]] = right;
            maxLength = Math.Max(maxLength, right - left + 1);
        }

        return maxLength;
    }

    static void Main(string[] args)
    {
        string s1 = "abcabcbb";
        string s2 = "bbbbb";
        string s3 = "pwwkew";

        Console.WriteLine("length of longest substring in '" + s1 + "': " + length(s1));
        Console.WriteLine("length of longest substring in '" + s2 + "': " + length(s2));
        Console.WriteLine("length of longest substring in '" + s3 + "': " + length(s3));
    }
}

File: example2.java
import java.util.HashMap;
import java.util.Map;

public class Main {
    public static int lengthOfLongestSubstring(String s) {
        int maxLength = 0;
        int left = 0;
        Map<Character, Integer> lastIndex = new HashMap<>();

        for (int right = 0; right < s.length(); right++) {
            if (lastIndex.containsKey(s.charAt(right)) && lastIndex.get(s.charAt(right)) >= left) {
                left = lastIndex.get(s.charAt(right)) + 1;
            }

            lastIndex.put(s.charAt(right), right);
            maxLength = Math.max(maxLength, right - left + 1);
        }

        return maxLength;
    }

    public static void main(String[] args) {
        String s1 = "abcabcbb";
        String s2 = "bbbbb";
        String s3 = "pwwkew";

        System.out.println("Length of longest substring in '" + s1 + "': " + lengthOfLongestSubstring(s1));
        System.out.println("Length of longest substring in '" + s2 + "': " + lengthOfLongestSubstring(s2));
        System.out.println("Length of longest substring in '" + s3 + "': " + lengthOfLongestSubstring(s3));
    }
}
